sdk/clientcore/core/src/main/java/io/clientcore/core/utils/CoreUtils.java

```diff
@@ -510,10 +510,11 @@ private static SerializationFormat checkForKnownEncoding(String contentType) {
      * @param data The BinaryData to decode.
      * @param serializer The serializer to use.
      * @param returnType The type of the ParameterizedType return value.
+     * @param <T> The decoded value type.
      * @return The decoded value.
      * @throws CoreException If the deserialization fails.
      */
-    public static Object decodeNetworkResponse(BinaryData data, ObjectSerializer serializer,
+    public static <T> T decodeNetworkResponse(BinaryData data, ObjectSerializer serializer,
```

Comment:
Why did we change this to `T`? We are not using it anywhere in the method.

Reply:
It's used in code generated with annotation processor. This was done to have the return match the assignment in return.